tests/iso-kargs.sh

```diff
@@ -21,94 +21,113 @@ cd "${tmpd}"
+    # the default kargs size to get the size of the overflow embed area.
+    local embed_size=$(coreos-installer iso kargs show --header "${iso}" | jq .length)
+    local embed_default_kargs_size=$(coreos-installer iso kargs show --default "${iso}" | wc -c)
+    local embed_usable_size=$((${embed_size} - ${embed_default_kargs_size} - 1))
+
+    local long_karg=$(printf '%*s' $((embed_usable_size)) | tr ' ' "k")
+    coreos-installer iso kargs modify -a "${long_karg}" "${iso}"
+    coreos-installer iso kargs show "${iso}" | grep -q " ${long_karg}\$"
+    coreos-installer iso kargs reset "${iso}"
+    long_karg=$(printf '%*s' $((embed_usable_size + 1)) | tr ' ' "k")
+    if coreos-installer iso kargs modify -a "${long_karg}" "${iso}" 2>err.txt; then
+        fatal "Was able to put karg longer than area?"
+    fi
+    grep -q 'kargs too large for area' err.txt
+
+    # Test `reset`.
+    coreos-installer iso kargs modify -a foobar "${iso}"
+    rm "${out_iso}"
+    coreos-installer iso kargs reset "${iso}" -o "${out_iso}"
+    hash=$(digest "${out_iso}")
+    if [ "${orig_hash}" != "${hash}" ]; then
+        fatal "Hash doesn't match original hash: ${hash} vs ${orig_hash}"
+    fi
+    coreos-installer iso kargs reset "${iso}" -o - > "${out_iso}"
+    hash=$(digest "${out_iso}")
+    if [ "${orig_hash}" != "${hash}" ]; then
+        fatal "Hash doesn't match original hash: ${hash} vs ${orig_hash}"
+    fi
+    coreos-installer iso kargs reset "${iso}"
+    hash=$(digest "${iso}")
+    if [ "${orig_hash}" != "${hash}" ]; then
+        fatal "Hash doesn't match original hash: ${hash} vs ${orig_hash}"
+    fi
+
+    # Check modification against expected ground truth.
+    coreos-installer iso kargs modify -a foobar=val "${iso}"
+    local offset=$(coreos-installer iso kargs show --header "${iso}" | jq -r .kargs[0].offset)
+    local length=$(coreos-installer iso kargs show --header "${iso}" | jq -r .kargs[0].length)
+    local expected_args="$(coreos-installer iso kargs show --default "${iso}") foobar=val"
+    local expected_args_len="$(echo -n "${expected_args}" | wc -c)"
+    local filler="$(printf '%*s' $((${length} - ${expected_args_len} - 1)) | tr ' ' '#')"
+    if ! echo -en "${expected_args}\n${filler}" | cmp -s <(dd if=${iso} skip=${offset} count=${length} bs=1) -; then
+        fatal "Failed to manually round-trip kargs"
+    fi
+
+    # Clean up
+    coreos-installer iso kargs reset "${iso}"
+}
+
+echo "============== Both headers =============="
+run_tests
+
+echo "============== Only JSON =============="
+dd if=/dev/zero of="${iso}" seek=32672 count=8 bs=1 conv=notrunc status=none
+run_tests
+
+echo "============== Only System Area =============="
+echo -n "coreKarg" | dd of="${iso}" seek=32672 bs=1 conv=notrunc status=none
+echo -n "X" | dd of="${iso}" seek=$(grep --byte-offset --only-matching --text 'KARGS.JSO;1' "${iso}" | cut -f1 -d:) bs=1 conv=notrunc status=none
```

Comment:
So, this is overwriting the name field in the directory record to be `XARGS.JSON;1`? Feels like it's worth a comment.